skbio/io/tests/test_registry.py

```diff
@@ -1552,13 +1562,21 @@ def writer(obj, fh):
         self.registry.write(obj, format='format1', into=fp)
 
         with io.open(fp, mode='rb') as fh:
-            self.assertEqual(b'\xa7A\xa6n\n', fh.read())
+            self.assertEqual(b"\xa7A\xa6n\n",
+                             fh.read()
+                             .decode(encoding="big5")
+                             .replace("\r\n", "\n")
+                             .encode(encoding="big5"))
 
         self._expected_encoding = 'utf8'
         self.registry.write(obj, format='format1', into=fp, encoding='utf8')
 
         with io.open(fp, mode='rb') as fh:
-            self.assertEqual(b'\xe4\xbd\xa0\xe5\xa5\xbd\n', fh.read())
+            self.assertEqual(b"\xe4\xbd\xa0\xe5\xa5\xbd\n",
```

Comment:
What if you don't decode/encode, but directly do `fh.read().replace(b"\r\n", b"\n")`?